streaming/src/main/scala/org/apache/spark/streaming/scheduler/JobGenerator.scala

```diff
@@ -243,6 +243,10 @@ class JobGenerator(jobScheduler: JobScheduler) extends Logging {
     // Example: BlockRDDs are created in this thread, and it needs to access BlockManager
     // Update: This is probably redundant after threadlocal stuff in SparkEnv has been removed.
     SparkEnv.set(ssc.env)
+
+    // Enable "spark.checkpoint.recursive" to make sure that all RDDs marked with the checkpoint
+    // flag are all checkpointed to avoid the stack overflow issue. See SPARK-6847
+    ssc.sparkContext.setLocalProperty("spark.checkpoint.recursive", "true")
```

Comment:
`localProperties` uses `InheritableThreadLocal` so they will be inherited by child threads.

However, looks it's better not to depend on this implicit assumption and just set it explicitly.